assets/src/eejs/validators/instance-of.js

```diff
@@ -7,12 +7,17 @@
  * @param {string} expectedName
  * @return {boolean} True means the object is an instance of expectedName.
  */
-export function instanceOf( objectInstance, expectedName ) {
-	if ( ! objectInstance ||
-		! objectInstance.constructor ||
-		! objectInstance.constructor.name
-	) {
+export function instanceOf(objectInstance, expectedName) {
+	if (!objectInstance) {
 		return false;
 	}
-	return objectInstance.constructor.name === expectedName;
+	if (objectInstance.constructor) {
+		if (objectInstance.constructor.name && objectInstance.constructor.name === expectedName) {
+			return true;
+		}
+		if (objectInstance.constructor.displayName && objectInstance.constructor.displayName === expectedName) {
+			return true;
+		}
+	}
+	return objectInstance.hasOwnProperty('displayName') && objectInstance.displayName === expectedName;
```

Comment:
could have done this much more compact but it really reduced the readability so i split things up like this to make it easier to grok

Reply:
👍